### PR TITLE
Rationalise module size handling

### PR DESCRIPTION
**Pull request by 3djc**
This attempt to clean a currently 'unclean' situation regarding external module size : depending on radio and/or module type, the list of available external module varies greatly, sometime without apparent logic.

This is built on the following assumptions:
- display only the module that will work on a radio without using adapters or modding the radio
- leave the possibility through compilation option to choose what to display

The modules fall into 3 groups:
A) both : those are the module that exist both in JR or small size, and that are accssed by the radio in the exact same way. Those will appear by default in all radio. This includes:
CRSF/ELRS
MULTI
GHOST
PPM
SBUS

B) those existing only on JR size, and been showed by default only in radio with JR bay
XJT
R9M
R9M ACCESS
DSM2
Lemon DSMP

C) those existing only in small size, and been showed by default only in radio with small bay
XJT Lite
R9M Lite
R9M Pro

### CODE REVIEW

**Comment by mha1** [comment]
Hi, good idea!

radio/src/targets/taranis/hal.h line 1780: think you accidentally removed a character at the end of the comment. PD.0 -> PD.08

**Comment by pfeerick** [comment]
LemonRX DSMP is indeed only JR Micro only form factor AFAIK. And I only knew of DSM2 modules in same form factor, but I've not actively followed the Nano form factor. 

A second part of the Nano assumption is that XJT Lite and R9M Lite (I don't know about the R9M Pro) will *NOT* work with any JR Micro adapter that I know of due to the use of the extra UART pins for ACCESS. I don't know of any Nano to JR Micro adapter that can work with that ;)

File: radio/src/targets/taranis/CMakeLists.txt [hunk 1]
```diff
@@ -173,7 +179,7 @@ elseif(PCB STREQUAL X7)
     set(CLI ON CACHE BOOL "Enable CLI")
   elseif(PCBREV STREQUAL TLITEF4)
     set(DEFAULT_INTERNAL_MODULE MULTIMODULE CACHE STRING "Default internal module")
-    set(INTERNAL_MODULE_SERIAL YES)
```
Comment (pfeerick) [inline radio/src/targets/taranis/CMakeLists.txt:176]: Was `INTERNAL_MODULE_SERIAL` supposed to be knocked off here?

Comment (3djc) [reply radio/src/targets/taranis/CMakeLists.txt:176]: No sir ! (thx for the catch)

**Comment by 3djc** [comment]
I think all comment so far taken into account

**Comment by elecpower** [comment]
@3djc do you want me to knock up a Companion side PR?
My thought is a an External Module Bay drop down list (None, Small, Standard) in Radio Profile Settings to cater for default and custom builds. When creating a new profile the initial value will be the radio compile default.
This value then used for Model Setup External Radio Module list of available Protocols.

**Comment by 3djc** [comment]
> @3djc do you want me to knock up a Companion side PR?
> My thought is a an External Module drop down list (None, Small, Standard) in Radio Profile Settings to cater for default and custom builds. When creating a new profile the initial value will be the radio compile default.
> This value then used for Model Setup External Radio Module list of available Protocols.

That would be really cool, but I think it should really be two options (JR and small) where you can tick none, one or both

**Comment by elecpower** [comment]
> That would be really cool, but I think it should really be two options (standard/JR and small) where you can tick none, one or both.

I assumed the choices would be mutually exclusive and thus the list. Also tick boxes might be a bit confusing to newbies and prone to being changed without understanding consequences.

This choice is per radio profile so you can have multiple profiles for say a TX16S with different external modules. Why not a list of None, JR, Small, Custom? IMO Standard implies factory supplied thus why I propose JR without Standard otherwise some may change Small to JR/Standard assuming the default is wrong and then cannot understand why their protocol is missing when configuring a model.

> If that can help, I can do a complete list of radio with their default

That would be very helpful, thanks.

**Comment by 3djc** [comment]
A list with none/std/small/both would work too. Yes, like many other choices, that break things when wrong :(. I think 99.9%  of users will use default.
- none is for case you need more eeprom, or you have your radio in a tray and know you will not be using an external module (or your radio defualt is none)
- both are for those who own adapters and use both module size
- changing default (small to JR or JR to small): I have see users modding and prin,ting alternate case to permanetly change the type of module their radio uses (typical is Zorro modded for JR modules, but it tend to become less frequent with nice namo modules comming out).

Will do the list.

**Comment by elecpower** [comment]
Thanks for the list

**Comment by pfeerick** [comment]
I just noticed in the list at https://github.com/EdgeTX/edgetx/pull/3624#issuecomment-1565235178 that the 
x9d+2019 is listed as having a small module, when in fact it has a JR Micro / Standard module bay. Working on the assumption that this was a mistake, I've changed that - just change it back if it was done due to an edge case or something.

**Comment by 3djc** [comment]
> I just noticed in the list at [#3624 (comment)](https://github.com/EdgeTX/edgetx/pull/3624#issuecomment-1565235178) that the x9d+2019 is listed as having a small module, when in fact it has a JR Micro / Standard module bay. Working on the assumption that this was a mistake, I've changed that - just change it back if it was done due to an edge case or something.
> 
Sorry my bad

> edit: As a site note, it would be nice to be able to filter the UI list even tighter... i.e. can the TX12MK2 even drive a R9M ACCESS module? Same applies for some other radios/form factors.
Yes it can, it needs a full uart, which it does have, but i'm gonna check how well (or not) it is actually handled

**Comment by 3djc** [comment]
Handling is right imho : https://github.com/EdgeTX/edgetx/blob/8325f722c4572fad856b1a07da7d154dae23084e/radio/src/gui/gui_common.cpp#L907

**Pull request merged**